File: 388-longest-absolute-file-path/longest-absolute-file-path.java
class Solution {
    public int lengthLongestPath(String input) {

        int maxLen =  0;
        String [] lines = input.split("\n");
        Map<Integer,Integer> pathlen = new HashMap<>();
        pathlen.put(0,0);

        for(String line: lines) {
            // String name = line.strip("\t"); --> doesnt work
            // removes tabs
            String name = line.replaceAll("\t", "");

            // Removes tab from start or end
            // String name = line.replaceAll("^\t+|\t+$", "");
            Integer depth = line.length() - name.length();

            if (name.contains(".")){
                maxLen = Math.max(maxLen, pathlen.get(depth) + name.length());
            }
            else {
                Integer val = pathlen.get(depth) + name.length() + 1;
                pathlen.put(depth+1, val);
            }
        
        }

        return maxLen;
    }
}

File: 388-longest-absolute-file-path/longest-absolute-file-path.py
class Solution:
    def lengthLongestPath(self, input: str) -> int:
        maxLen = 0
        pathlen = {0:0}

        for line in input.split("\n"):
            name = line.lstrip("\t")
            depth = len(line) - len(name)

            if "." in name: # end of the dirs, we found a file:
                 # a file
                maxLen = max(maxLen, pathlen[depth] + len(name))
            else:
                # either dir
                pathlen[depth + 1] = pathlen[depth] + len(name) + 1
            # new subdire = previous_subdir + len(subdr) + 1 --> "/"
        return maxLen